python/scrutiny/server/device/request_generator/device_searcher.py

```diff
@@ -37,17 +37,25 @@ def reset(self):
         self.found_device_timestamp = time.time()
         self.started = False
         self.found_device = None
-        self.found_device_display_name = ''
 
-    def get_found_device(self):
-        return self.found_device
+    def device_found(self):
+        return self.found_device is not None
 
-    def get_found_device_ascii(self):
-        if self.found_device is not None:
-            return binascii.hexlify(self.found_device).decode('ascii')
+    def get_device_firmware_id(self):
+        if self.device_found():
+            return self.found_device['firmware_id']
```

Comment:
missing else with default value return statement.
```suggestion
    def get_device_firmware_id(self):
        if self.device_found():
            return self.found_device['firmware_id']
        return None
```